lib/authorizer/aws4.js

```diff
@@ -6,13 +6,16 @@ var aws4 = require('aws4'),
     // Ideally, they're not supposed to be a part of the signature calculation
     NON_SIGNABLE_HEADERS = ['cache-control', 'postman-token'];
 
+/**
+ * @implements {AuthHandlerInterface}
+ */
 module.exports = {
     /**
      * Initializes a item (fetches all required parameters, etc) before the actual authorization step.
      *
      * @param {AuthInterface} auth
      * @param {Response} response
-     * @param done
+     * @param {authInitHookCallback} done
```

Comment:
Possible inconsistent casing? `AuthInitHookCallback`

Reply:
it's not a class and
it will not be instantiated

Reply:
To group all the hooks in one place, adding namespace to all the hooks 
so now it will be `AuthHandlerInterface~authInitHookCallback`